### PR TITLE
needimport supports extra options and links

### LINKED ISSUE #227: needimport ignores all links between needs
I have a proper `needs.json` containing some valid needs that are linked to each other. Now when I "needimport" this file, all needs are imported, but without links. I couldn't find a hint in the documentation, that specifies this behavior: https://sphinxcontrib-needs.readthedocs.io/en/latest/directives/needimport.html

Probably this is a bug? Or are there any other way to achieve this?

### PR DESCRIPTION
**Pull request by danwos**
* extra options and links are imported
* pre/post/template can be defined by needimport
* same for style and layout

Fixes #227

### CODE REVIEW

File: sphinxcontrib/needs/api/need.py [hunk 1]
```diff
@@ -370,7 +373,13 @@ def _read_in_links(links_string):
...
     links = []
     if links_string:
-        for link in re.split(";|,", links_string):
+        # Check if links_string is really a string, otherwise it will be a list, which can be used
+        # without modifications
+        if isinstance(links_string, str):
+            link_list = re.split(";|,", links_string)
+        else:
+            link_list = links_string
+        for link in link_list:
```
Comment (danieleades) [inline sphinxcontrib/needs/api/need.py:382]: do you want to add a type annotation to this function?

```python
def _read_in_links(links_string: Union[str, List[str]]):
    ...
```
?

File: sphinxcontrib/needs/directives/needimport.py [hunk 2]
```diff
@@ -120,15 +119,49 @@ def run(self):
...
-        content = template.render(needs_list=needs_list, hide=hide, id_prefix=id_prefix)
-        self.state_machine.insert_input(content.split('\n'),
-                                        self.state_machine.document.attributes['source'])
+        # template_location = os.path.join(os.path.dirname(os.path.abspath(__file__)), "needimport_template.rst")
+        # with open(template_location, "r") as template_file:
+        #     template_content = template_file.read()
+        # template = Template(template_content)
+        # content = template.render(needs_list=needs_list, hide=hide, id_prefix=id_prefix)
+        # self.state_machine.insert_input(content.split('\n'),
+        #                                 self.state_machine.document.attributes['source'])
```
Comment (danieleades) [inline sphinxcontrib/needs/directives/needimport.py:128]: if this is no longer used, should it just be deleted?

File: sphinxcontrib/needs/api/need.py (hunk 1 above)
Comment (danwos) [reply sphinxcontrib/needs/api/need.py:382]: Yes, will add it.
Sorry I'm not so familiar with type annotations (till now) :)

File: sphinxcontrib/needs/directives/needimport.py (hunk 2 above)
Comment (danwos) [reply sphinxcontrib/needs/directives/needimport.py:128]: Ohh yes.
Will add you next time as reviewer.
Thanks.

**Pull request merged**